spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -778,6 +780,7 @@ def stub_auth
           authn_context: Saml::Idp::Constants::LOA1_AUTHN_CONTEXT_CLASSREF,
           service_provider: 'http://localhost:3000',
           idv: false,
+          finish_profile: false,
```

Comment:
What about the `finish_profile: true` scenario? Maybe also add a feature spec in `spec/features/saml/loa3_sso_spec.rb` to test this whole flow?